src/adapters/angle-protocol/products/savings/angleProtocolSavingsAdapter.ts

```diff
@@ -0,0 +1,387 @@
+      name: 'AngleProtocol',
+      description: 'AngleProtocol defi adapter',
+      siteUrl: 'https://angle.money',
+      iconUrl:
+        'https://raw.githubusercontent.com/AngleProtocol/angle-assets/main/02%20-%20Logos/02%20-%20Logo%20Only/angle-only-fill-coral.png',
+      positionType: PositionType.Staked,
+      chainId: this.chainId,
+      productId: this.productId,
+      assetDetails: {
+        type: AssetType.StandardErc20,
+      },
+    }
+  }
+
+  @CacheToFile({ fileKey: 'protocol-tokens' })
+  async buildMetadata() {
+    const contractAddresses: Record<
+      Stablecoin,
+      Partial<Record<Chain, string>>
+    > = {
+      [Stablecoin.stEUR]: {
+        [Chain.Ethereum]: getAddress(
+          '0x004626A008B1aCdC4c74ab51644093b155e59A23',
+        ),
+        [Chain.Polygon]: getAddress(
+          '0x004626A008B1aCdC4c74ab51644093b155e59A23',
+        ),
+        [Chain.Bsc]: getAddress('0x004626A008B1aCdC4c74ab51644093b155e59A23'),
+        [Chain.Arbitrum]: getAddress(
+          '0x004626A008B1aCdC4c74ab51644093b155e59A23',
+        ),
+        [Chain.Optimism]: getAddress(
+          '0x004626A008B1aCdC4c74ab51644093b155e59A23',
+        ),
+        [Chain.Avalanche]: getAddress(
+          '0x004626A008B1aCdC4c74ab51644093b155e59A23',
+        ),
+        [Chain.Base]: getAddress('0x004626A008B1aCdC4c74ab51644093b155e59A23'),
+      },
+      [Stablecoin.stUSD]: {
+        [Chain.Ethereum]: getAddress(
+          '0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776',
+        ),
+        [Chain.Polygon]: getAddress(
+          '0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776',
+        ),
+        [Chain.Linea]: getAddress('0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776'),
+        [Chain.Bsc]: getAddress('0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776'),
+        [Chain.Arbitrum]: getAddress(
+          '0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776',
+        ),
+        [Chain.Optimism]: getAddress(
+          '0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776',
+        ),
+        [Chain.Avalanche]: getAddress(
+          '0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776',
+        ),
+        [Chain.Base]: getAddress('0x0022228a2cc5E7eF0274A7Baa600d44da5aB5776'),
+      },
+    }
+
+    // load the contract based on this.chainId
+    const savingsContract: Map<Stablecoin, Savings> = new Map()
+    Object.keys(contractAddresses).map((key) => {
+      if (!contractAddresses[key as Stablecoin][this.chainId]) return
+      const contract = Savings__factory.connect(
+        contractAddresses[key as Stablecoin][this.chainId]!,
+        this.provider,
+      )
+      savingsContract.set(key as Stablecoin, contract)
+    })
+
+    const underlyingTokenAddresses: Map<Stablecoin, string> = new Map()
+    await Promise.all(
+      Array.from(savingsContract).map(async ([key, contract]) => {
+        underlyingTokenAddresses.set(key, await contract.asset())
+      }),
+    )
+
+    const protocolTokens: Map<Stablecoin, Erc20Metadata> = new Map()
```

Comment:
I find this name confusing

Reply:
It should remains protocolTokens IMO as they are the tokens that are going to get returned in `getProtocolTokens` function.

Tell me if you have another naming idea